src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { SetStateAction, useEffect, useState } from 'react';
+import { CustomError } from '../../types/Error';
+import { LoadingTodo } from '../../types/LoadingTodo';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from './TodoItem';
+
+type TodoListProps = {
+  renderedTodos: Todo[];
+  setRenderedTodos: React.Dispatch<SetStateAction<Todo[]>>;
+  initialTodos: Todo[];
+  setInitialTodos: React.Dispatch<SetStateAction<Todo[]>>;
+  setErrorMessage: React.Dispatch<SetStateAction<CustomError>>;
+  loadingTodo: LoadingTodo;
+  setLoadingTodo: React.Dispatch<SetStateAction<LoadingTodo>>;
+  loadingTodos: LoadingTodo[];
+};
+
+export const TodoList: React.FC<TodoListProps> = ({
+  renderedTodos,
+  setRenderedTodos,
+  initialTodos,
+  setInitialTodos,
+  setErrorMessage,
+  loadingTodo,
+  setLoadingTodo,
+  loadingTodos,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    return renderedTodos !== initialTodos
+      ? setTodos(renderedTodos)
+      : setTodos(initialTodos);
+  }, [renderedTodos, initialTodos]);
```

Comment:
please, feel free to ask in the chat how to rewrite that using only one state for todos